### ISSUE TITLE
Request to WSS of ripple clio leads to clash

### ISSUE THREAD

**Issue by monistillbear** [XRPLF/xrpl.js#2565]
Hey guys, I get this uncaught exception. It doesn't happen always, but often enough to be an issue to the project. It happens to different requests, from what I noticed, mostly to _Client.autofill()_.

xrpl.js library version is 2.12.0
ws library version is 8.13.0

Tried updating to:
xrpl.js -> 2.13.0
ws -> 8.14.0
This did not help.

The main issue is that the problem seems to be outside of my own code.

This didn't happen for a while after the code has been written, not even during the load testing and appeared recently.

The issue is similar to https://github.com/XRPLF/xrpl.js/issues/1338 but it has been marked as 'resolved', closed and I was suggested to open a new one.

```
error - uncaughtException: Error [ERR_UNHANDLED_ERROR]: Unhandled error. ('badMessage')
    at new NodeError (node:internal/errors:400:5)
    at Client.emit (node:events:502:17)
    at Connection.<anonymous> (/node_modules/xrpl/dist/npm/client/index.js:94:18)
    at Connection.emit (node:events:513:28)
    at Connection.onMessage (/node_modules/xrpl/dist/npm/client/connection.js:240:26)
    at WebSocket.<anonymous> (/node_modules/xrpl/dist/npm/client/connection.js:255:53)
    at WebSocket.emit (node:events:513:28)
    at Receiver.receiverOnMessage (/node_modules/ws/lib/websocket.js:1184:20)
    at Receiver.emit (node:events:513:28)
    at Receiver.dataMessage (/node_modules/ws/lib/receiver.js:541:14)
    at Receiver.getData (/node_modules/ws/lib/receiver.js:459:17)
    at Receiver.startLoop (/node_modules/ws/lib/receiver.js:158:22)
    at Receiver._write (/node_modules/ws/lib/receiver.js:84:10)
    at writeOrBuffer (node:internal/streams/writable:392:12)
    at _write (node:internal/streams/writable:333:10)
    at Writable.write (node:internal/streams/writable:337:10)
    at TLSSocket.socketOnData (/node_modules/ws/lib/websocket.js:1278:35)
    at TLSSocket.emit (node:events:513:28)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23) {
  digest: undefined
}
```

**Comment by ckniffen** [comment]
Does this only happen with clio?  Have you tried using xrplcluster.com?

**Comment by mvadari** [comment]
Please include a code snippet that we can use to replicate.

**Comment by monistillbear** [comment]
@ckniffen thank you very much!
Tried to run the same code while connected to public test network wss (wss://s.altnet.rippletest.net:51233/).
After everything kept being successful, we looked into clio.
The issue was that **non-whitelisted ip addresses in clio have a rate limit**. 
If limit is exceeded, it causes this uncaught exception.
The code handled load of several thousands of transactions and everything was fine afterwards.

Thanks to everybody for your input!
